examples/shallow/execute.py

```diff
@@ -31,60 +31,72 @@
 #
 # Author R. Ford STFC Daresbury Lab.
 #
-''' '''
+'''An example use of melody in which we search a set of compiler
+flags to find the best runtime for the shallow benchmark'''
 
 from melody.inputs import create_input
 
+
 def execute(option):
-    ''' xxx '''
+    '''A script that melody calls with each valid set of options. This
+    script runs the required code and returns the results.'''
 
     namelist_option = []
     makefile_option = []
     flags = ""
     for entry in option:
         key = entry.keys()[0]
         if key == "Problem Size":
-            namelist_option.append({"SIZE":entry[key]})
+            namelist_option.append({"SIZE": entry[key]})
         elif key == "F90":
             makefile_option.append(entry)
         else:
             flags += entry[key] + " "
-    makefile_option.append({"F90FLAGS":flags})
-            
-    namelist = create_input(namelist_option, "namelist", template_location="templates")
+    makefile_option.append({"F90FLAGS": flags})
+
+    namelist = create_input(namelist_option, "namelist",
+                            template_location="templates")
 
-    makefile_include = create_input(makefile_option, "Makefile.include", template_location="templates")
+    makefile_include = create_input(makefile_option, "Makefile.include",
+                                    template_location="templates")
 
     benchmark_base = "/home/rupert/proj/GOcean/shallow_benchmark"
 
     # save the input files in the appropriate place
     location = benchmark_base + "/original/namelist"
-    f = open(location, 'w')
-    f.write(namelist)
-    f.flush()
+    my_file = open(location, 'w')
+    my_file.write(namelist)
+    my_file.flush()
 
     location = benchmark_base + "/common/Makefile.include"
-    f = open(location, 'w')
-    f.write(makefile_include)
-    f.flush()
+    my_file = open(location, 'w')
+    my_file.write(makefile_include)
+    my_file.flush()
 
     # compile shallow if required
     #import os
     #os.chdir("/home/rupert/proj/GOcean/shallow_benchmark/original")
+    base_path = "/home/rupert/proj/GOcean/shallow_benchmark/original"
```

Comment:
I've added the required benchmark code to the repository for the time being. I've tested it and it runs OK.